### ISSUE TITLE
opening_brace generates false positive on anonymous closure

### ISSUE THREAD

**Issue by idrougge** [realm/SwiftLint#2879]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`opening_brace` (which is enabled by default) triggers on an anonymous closure, e.g.:
```swift
func f(rect: CGRect) {
   if rect.origin.x > 0 {
      print(rect.origin)
   }
   { // Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
      let centre = CGPoint(x: rect.midX, y: rect.midY)
      print(centre)
   }()
}
```
### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.35.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
Homebrew

**Comment by heidiproske** [comment]
Another example that's _somewhat_ related:

```swift
func testButtonStyling() {
    let primaryButton = UIButton()
    primaryButton.frame = CGRect(x: 0, y: 0, width: 20, height: 20)
    assertTrueAfterStyling(comparables: [
        { primaryButton.backgroundColor == .brown },
        { primaryButton.layer.cornerRadius == (primaryButton.frame.height / 2) }
    ])
}

func assertTrueAfterStyling(comparables: [() -> Bool], file: StaticString = #file, line: UInt = #line) {
    XCTAssert(true)
}
```

The line calling assertTrueAfterStyling results in 2 warnings:
`Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)`

which I don't think should be triggered... 

You get the same errors if you try change the whitespace to this:
```swift
        assertTrueAfterStyling(comparables: [
            {
                primaryButton.backgroundColor == .brown
            },
            {
                primaryButton.layer.cornerRadius == (primaryButton.frame.height / 2)
            }
        ])
```

And if you try change it to listen to the rule (where the opening brace goes on same line as the declaration) like this:
```swift
        assertTrueAfterStyling(comparables: [ {
                primaryButton.backgroundColor == .brown
            }, {
                primaryButton.layer.cornerRadius == (primaryButton.frame.height / 2)
            }
        ])
```

then you get a warning re `Collection Element Alignment Violation: All elements in a collection literal should be vertically aligned (collection_alignment)`

**Comment by idrougge** [comment]
I worked around the issue by inserting a comment explaining the bug above the anonymous closure.

**Comment by csvrcek** [comment]
@Pratix98 and I, first-time contributors, would like to contribute to this issue if nobody else was going to!

**Comment by csvrcek** [comment]
Looks like there is a pr up to fix this (see the bugfix-2879 branch), so @Pratix98 and I will move on from this issue

**Comment by a-25** [comment]
@csvrcek: sorry, but I have fixed that already :)
